var/spack/repos/builtin/packages/scine-qcmaquis/package.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class ScineQcmaquis(CMakePackage):
+    """Scine QCMaquis DMRG Solver"""
+
+    homepage = "https://scine.ethz.ch/download/qcmaquis"
+    git = "https://github.com/qcscine/qcmaquis.git"
+
+    maintainers("adam-grofe")
+
+    version("master", branch="master")
+    version("3.1.3", branch="release-3.1.3")
+    version("3.1.2", branch="release-3.1.2")
+    variant(
+        "blas",
+        values=("openblas", "mkl"),
+        default="openblas",
+        description="Which blas library to use.",
+    )
+    variant(
+        "symmetries",
+        default="TwoU1;SU2U1;SU2U1PG;TwoU1PG",
+        description='Wave functions symmetries to compile (e.g. "TwoU1;SU2U1;SU2U1PG;TwoU1PG")',
+    )
+    variant(
+        "build_tests",
+        default=True,
+        description="Whether to build unit tests using gtest and gmock",
+    )
+
+    root_cmakelists_dir = "dmrg"
+
+    depends_on("hdf5~mpi")
+    depends_on("lapack")
+    depends_on("openblas+ilp64 threads=openmp", when="blas=openblas")
+    depends_on("intel-oneapi-mkl", when="blas=mkl")
+    depends_on("gsl")
+    depends_on("boost+program_options+filesystem+system+thread+serialization")
+    depends_on("googletest+gmock", when="build_tests=True")
+
+    def cmake_args(self):
+        args = []
+        build_symm = str(self.spec.variants["symmetries"].value)
+        if build_symm:
+            args.extend(["-DBUILD_SYMMETRIES={}".format(build_symm)])
+        build_tests = str(self.spec.variants["build_tests"].value)
+        if build_tests:
+            args.extend(["-DQCMAQUIS_TESTS=On"])
```

Comment:
This isn't doing what you want.  

Try `spack install scene-qcmaquis~build_tests` (so disable the variant) and you'll see that the option is translated to `'-DQCMAQUIS_TESTS=On'`.

Reply:
After changing the cmake args function to use `define_from_variant`, this bug was fixed. I verified it myself in a clean environment.